src/DataStructures.jl

```diff
@@ -112,4 +112,5 @@ module DataStructures
     include("splay_tree.jl")
 
     include("deprecations.jl")
+    @deprecate top(h::BinaryHeap) = first(h)
```

Comment:
This is not sufficient since for other heaps the deprecation didn't work either:
```suggestion
@deprecate top(x) = first(h)
```
Probably should go into `src/deprecations.jl` though.